backends/ebpf/runtime/ebpf_runtime.c

```diff
@@ -106,37 +209,23 @@ int main(int argc, char **argv) {
         BPF_OBJ_PIN(current, current->name);
         current++;
     }
-    /* Open and read pcap file */
-    in_handle = pcap_open_offline(input_pcap, errbuf);
-    if (in_handle == NULL) {
-        pcap_perror(in_handle, "Error: Failed to read pcap file ");
-        return EXIT_FAILURE;
-    }
-    /* Create the output file. */
-    char *in_file_name = strdup(input_pcap);
-    char *in_dir = dirname(in_file_name);
-    int in_dir_len = strlen(in_dir);
-    /* length of the input directory plus the filename max length */
-    char *out_file_name = calloc(in_dir_len + FILE_NAME_MAX , 1);
-    snprintf(out_file_name, in_dir_len + FILE_NAME_MAX, "%s/pcap0_out.pcap", in_dir);
-
-    /* Open the output file */
-    out_handle = pcap_dump_open(in_handle, out_file_name);
-    if (out_handle == NULL) {
-        pcap_perror(in_handle, "Error: Failed to create pcap output file ");
-        pcap_close(in_handle);
-        free(out_file_name);
-        return EXIT_FAILURE;
-    }
 
+#ifdef CONTROL_PLANE
     /* Set the default action for the userspace hash tables */
-    initialize_tables();
-    generated_init();
-    int ret = process_packets(in_handle, out_handle);
-    if (ret == -1)
-        pcap_perror(in_handle, "Error: Failed to parse ");
-    pcap_close(in_handle);
-    pcap_dump_close(out_handle);
-    free(out_file_name);
+    init_tables();
+    /* Run all commands specified in the control file */
+    setup_control_plane();
+#endif
+
+    launch_runtime(pcap_name, num_pcaps);
+
+    /* Delete all the remaining tables in the user program */
```

Comment:
For symmetry you should move this code into a function delete_tables.
Also, this code is only needed for the user-space testing. Is this file used when doing kernel-level testing too?